### PR TITLE
High-danger plants like deathnettles now require botanical gloves to hold instead of any glove type

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

- Deathnettles, Novaflowers, and Roses now require botanical leather gloves to hold, instead of just any gloves.
- Bluespace Tomatos and Killer Tomatos now have backfire effects for picking them up without gloves.
- Podpeople are naturally immune to the above and don't need leather gloves - I figured it made sense.
- Fixed a wrongfully typed path for plasmaman botanical leather gloves.
- Kills off a useless subtype of nettles that didn't have a reason to exist, for standardization.
- Moves the force initialization for novaflowers off of `on_juice` (???) to initialize.

## Why It's Good For The Game

So, it's a very common misconception that deathnettles and similar plants requires botanical leather gloves to yield. 

They do not - they just require you are wearing any gloves.

BUT, it is both appropriate thematically, as well as a slight nerf to botany, if they were forced to use their leather gloves to use the special plants. This makes botanical leather gloves the glove of choice for the botanist instead of insuls or anything else, which is what it should be.

## Changelog
:cl: Melbert
balance: Deathnettles, Novaflowers, and similar plants now require botanical leather gloves (or biosuits/hardsuits) to hold and use instead of any pair of gloves. Additionally, Novaflowers now deal their force to their wielder instead of 5 flat damage.
balance: Bluespace Tomatos and Killer Tomatos now require botanical gloves to hold and use safely, and can backfire on you if you aren't wearing them.
balance: Podpeople are naturally immune to plants harming them or backfiring on use.
fix: Corrected the typepath of Plasmaman botanical gloves.
code: Improves/modernizes nettle and novaflower code a bit.
/:cl:

### CODE REVIEW

File: code/modules/hydroponics/grown/tomato.dm [hunk 1]
```diff
@@ -147,3 +174,16 @@
...
+	. = ..()
+	if(!iscarbon(user))
+		return
+
+	var/mob/living/carbon/carbon_user = user
+	if(HAS_TRAIT(carbon_user, TRAIT_PLANT_SAFE))
+		return
+
+	if(prob(25))
+		to_chat(carbon_user, "<span class='danger'>\The [name] begins to growl and shake!</span>")
```
Comment (Mothblocks) [inline code/modules/hydroponics/grown/tomato.dm:188]:
```suggestion
		to_chat(carbon_user, "<span class='danger'>[src] begins to growl and shake!</span>")
```

Comment (MrMelbert) [reply code/modules/hydroponics/grown/tomato.dm:188]: image
*shrug

Comment (Mothblocks) [reply code/modules/hydroponics/grown/tomato.dm:188]: Try renaming it to `\improper killer-tomato`.

**Comment by rdragan** [comment]
How is making their weapons unusable by the rest of the crew a nerf to botany?  
Rogue botanists often die by their own weapons; if anything, this is a slight buff. This also makes it harder to gather up deathslip plants that the botanists might have left lying around the station.

**Comment by MrMelbert** [comment]
> How is making their weapons unusable by the rest of the crew a nerf to botany?
You can no longer mass-produce stuff like deathnettles for nukeops or cult, and if you want to use deathnettles yourself you can't use the insulateds you tided out of tech storage. Insulated gloves in comparison recently became very good for botanists because botanists don't care about being unable to use guns.

> This also makes it harder to gather up deathslip 
Slippery skin deathnettles are large and obvious, if someone's leaving them around you can gather them in a locker quite easily, use a broom, or just go print your own pair of leather gloves as the biogenerator can do that.

**Comment by ATH1909** [comment]
eh, I dunno, it seems kind of odd that people wearing captain's gloves, hardsuits, etc. would still get stung when picking up deathnettles. maybe give botanical gloves, captain's gloves, etc. the thickmaterial flag and check for that (on both your gloves and any arms-covering exosuits) instead?

**Comment by ATH1909** [comment]
also, shouldn't pierceimmune races like golems not trigger these plant effects? or should they, since they're directly touching them (in the case of golems, anyway)?

**Comment by KathyRyals** [comment]
I agree that cap gloves and hardsuits should make you immune to it.

**Comment by Mickyan** [comment]
Having very specific equipment that most jobs do not have immediate access to be the only hard counters to something is going to suck and it's the same reason why flashes have been reworked.

If you genuinely want this to be a botanist nerf you should apply this to attacking and throwing instead of pickup, otherwise this is effectively a nerf to insulated gloves and a buff to traitor botanists

**Comment by MrMelbert** [comment]
> Captain's gloves / hardsuits

I'd like there to be a choice between insulation and powerful plants for botanists, but hardsuits is valid and can be touched on in the future

> Pierce immune

Pierce immune species are immune to some of the plants like Nettles, this is already how the behavior works and it's remaining unchanged

> If you genuinely want this to be a botanist nerf you should apply this to attacking and throwing instead of pickup,

That is a valid point, I'll add the backfire effects to attacks as well, to avoid cheese.

**Comment by MrMelbert** [comment]
Nettles and novaflowers now also prevent you from attacking with them if you're not properly protected. I'm not thrilled with the implementation and it could probably be turned into an element but i'm open to suggestions.

**Comment by ExcessiveUseOfCobblestone** [comment]
you're going to have to explain very carefully how a glove that botany starts out with being the only way to handle the thing that botany has nearly sole access to is a nerf.

The only practical nerf for a botanist would be those that are using these exotic plants for food, because now the chef is going to get trolled unless they too wear the gamer gloves.

**Comment by MrMelbert** [comment]
(None of the "high risk" plants are used directly in any cooking recipes (with the exception of the base nettle, which is not that harmful without gloves and can be carried with trays/bags))

The main nerf is 2 part:
- Botanists cannot mass-produce stuff like Deathnettles (Liquid contents deathnettles) for the entire crew to use on large-scale station threats without also providing gloves (Note that the biogenerator can print leather gloves, so they're not entirely uncommon) for people to use. Deathnettles get up to esword damage so handing out bags of eswords is a bit silly, especially if they're liquid contents because they're busted as fuck the way acid works.

- Botanists have to choose between using other utility gloves (Insulated gloves they tide from tech storage round start when getting their chem dispenser, color gloves to disguise their fibers) and using high-danger plants like Deathnettles and Bluespace Tomatos, playing into the whole 'glove balance' game.

**Comment by MrMelbert** [comment]
I went ahead and made people wearing `THICKMATERIAL` clothing that covers the hands protect against dangerous plants to assist in cleanup.

**Comment by Tristrian** [comment]
Heya. I just tought about something concerning regarding balance : 
This makes "50 Deathneetles bag quick draw throw spam botanists" stronger if that prevents people to self-defense from picking up missed deathneetles to defend themselves (unless they have the thickmaterial).

Can you even catch the deathneetle and dropping it without receiving damage ?

**Comment by Mothblocks** [comment]
I don't think this is a nerf to botanist at all, I just think it makes sense and further follows what I believe of clothing being specifically useful to those whose job they belong to.

**Pull request merged**